database/README.md

```diff
@@ -4,29 +4,41 @@ database
 [![ISC License](http://img.shields.io/badge/license-ISC-blue.svg)](https://choosealicense.com/licenses/isc/)
 [![GoDoc](https://img.shields.io/badge/godoc-reference-blue.svg)](http://godoc.org/github.com/kaspanet/kaspad/database)
 
-Package database provides a block and metadata storage database.
+Package database provides a database for kaspad.
 
-Please note that this package is intended to enable kaspad to support different
-database backends and is not something that a client can directly access as only
-one entity can have the database open at a time (for most database backends),
-and that entity will be kaspad.
+Overview
+--------
+This package provides a database layer to store and retrieve data in a simple
+and efficient manner.
 
-When a client wants programmatic access to the data provided by kaspad, they'll
-likely want to use the [rpcclient](https://github.com/kaspanet/kaspad/tree/master/rpcclient)
-package which makes use of the [JSON-RPC API](https://github.com/kaspanet/kaspad/tree/master/docs/json_rpc_api.md).
+The current backend is ffldb, which makes use of leveldb, flat files, and strict
+checksums in key areas to ensure data integrity.
 
-The default backend, ffldb, has a strong focus on speed, efficiency, and
-robustness. It makes use of leveldb for the metadata, flat files for block
-storage, and strict checksums in key areas to ensure data integrity.
+Implementors of additional backends are required to implement the following interfaces:
 
-## Feature Overview
+DataAccessor
+------------
+This defines the common interface by which data gets accessed in a generic kaspad
+database. Both the Database and the Transaction interfaces (see below) implement it.
 
-- Key/value metadata store
-- Kaspa block storage
-- Efficient retrieval of block headers and regions (transactions, scripts, etc)
-- Read-only and read-write transactions with both manual and managed modes
-- Nested buckets
-- Iteration support including cursors with seek capability
-- Supports registration of backend databases
-- Comprehensive test coverage
+Database
+--------
+This defines the interface of a database that can begin transactions, open cursors,
+and close itself.
 
+Important: This is not part of the DataAccessor interface because the Transaction
+interface implements it as well. Were we to merge Database with DataAccessor,
+implementors of the Transaction interface would be forced to implement methods
+such as Begin and Close, which is undesirable.
+
+Transaction
+-----------
+This defines the interface of a generic kaspad database transaction.
+
+Note: Transactions provide data consistency over the state of the database as it was
+when the transaction started. There is NO guarantee that if one puts data into the
```

Comment:
"There is NO guarantee..."
If I remember correctly there's a guarantee that it won't happen.

Reply:
This comment refers to the database package, whereas the not-happening guarantee is specific to leveldb.